exla/lib/exla.ex

```diff
@@ -184,6 +176,45 @@ defmodule EXLA do
 
   @behaviour Nx.Defn.Compiler
 
+  @doc """
+  Sets the global defn options to the EXLA compiler with the
+  preferred client based on the available platforms.
+
+  This function is typically invoked at the top of scripts
+  and code notebooks which might be potentially executed from
+  multiple platforms. Do not invoke this function during runtime,
+  as it changes the `Nx.Defn` options globally. If you have a
+  specific client that you want to use throughout your project,
+  use configuration files instead:
+
+      config :nx, :default_defn_options, [compiler: EXLA, client: :cuda]
+
+  ## Examples
+
+      EXLA.set_preferred_defn_options([:tpu, :cuda, :rocm, :host])
+
+  The above will try to find the first client available and set
+  the `EXLA` compiler with the client as the compilers for `Nx.Defn`.
+  If no client is found, `EXLA` is not set as compiler at all,
+  therefore it is common to add `:host` as the last option.
+  """
+  def set_preferred_defn_options(clients) do
+    supported_platforms = EXLA.Client.get_supported_platforms()
+    all_clients = Application.fetch_env!(:exla, :clients)
+
+    chosen =
+      Enum.find(clients, fn client ->
+        client_config = all_clients[client]
+        client_platform = client_config[:platform] || :host
+        client_config && Map.has_key?(supported_platforms, client_platform)
+      end)
+
+    if chosen do
+      Nx.Defn.global_default_options(compiler: EXLA, client: chosen)
```

Comment:
`Nx.Defn.set_global_default_options` would be more consistent, but maybe it's too much?